internal/templates/templates.go

```diff
@@ -0,0 +1,44 @@
+package templates
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"html/template"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+//go:embed *.yaml
+var templates embed.FS
+
+func ResourceFromTemplate[T any, R any](t *T, name string, scheme *runtime.Scheme) (*R, error) {
+	resYaml, err := templates.ReadFile(fmt.Sprintf("%s.yaml", name))
+	if err != nil {
+		return nil, fmt.Errorf("could not read %s template: %v", name, err)
+	}
+	tmpl, err := template.New(name).Parse(string(resYaml))
+	if err != nil {
+		return nil, fmt.Errorf("could not parse %s template: %v", name, err)
+	}
+	var buf bytes.Buffer
+
+	err = tmpl.Execute(&buf, struct {
+		Values      *T
+		ExtraValues any
```

Comment:
Looks like I missed this?

Reply:
Right, I just removed it. Also the `r.Scheme` argument was unused in this function.